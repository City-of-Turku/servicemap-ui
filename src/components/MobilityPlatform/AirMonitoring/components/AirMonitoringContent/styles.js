const styles = (theme) => ({
  popupInner: {
    borderRadius: '3px',
    marginBottom: '0.4rem',
    marginLeft: '0.6rem',
    lineHeight: 1.2,
    overflowX: 'hidden',
  },
  buttonTransparent: {
    backgroundColor: '#fff',
    border: 'none',
    cursor: 'pointer',
    '&:hover': {
      color: 'rgba(84, 84, 84, 255)',
    },
  },
  button: {
    border: '1px solid gray',
    borderRadius: '5px',
    cursor: 'pointer',
    marginRight: theme.spacing(1.5),
  },
  buttonGray: {
    backgroundColor: '#ddd',
    border: 'none',
    borderRadius: '5px',
    color: '#fff',
    cursor: 'pointer',
    padding: theme.spacing(1),
  },
  buttonWhite: {
    backgroundColor: '#fff',
  },
  buttonActive: {
    backgroundColor: 'rgba(7, 44, 115, 255)',
    color: '#fff',
  },
  paddingWide: {
    padding: `${theme.spacing(0.5)} ${theme.spacing(1)}`,
  },
  paddingNarrow: {
    padding: theme.spacing(0.5),
  },
  widthMd: {
    width: '95%',
  },
  widthSm: {
    width: '87%',
  },
  contentHeader: {
    display: 'flex',
    flexDirection: 'row',
    marginTop: theme.spacing(0.5),
    marginBottom: theme.spacing(1.5),
    alignItems: 'flex-end',
    borderBottom: '2px solid gray',
    justifyContent: 'space-between',
  },
  headerSubtitle: {
    padding: '4px 0 5px',
    fontWeight: 'bold',
    marginBlockStart: theme.spacing(2),
    marginBlockEnd: theme.spacing(0.2),
  },
  dateContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    maxWidth: '32%',
  },
  container: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  parameterTypes: {
    display: 'flex',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    marginBottom: theme.spacing(0.75),
  },
  parameterTypeText: {
    fontWeight: 'bold',
    fontSize: '0.8rem',
    paddingTop: theme.spacing(0.5),
    paddingRight: theme.spacing(1),
  },
  buttonText: {
    fontSize: '0.75rem',
  },
  chartContainer: {
    margin: 0,
  },
  iconWrapper: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(7, 44, 115, 255)',
    color: '#fff',
    border: '1px solid gray',
    borderRadius: '5px',
    padding: theme.spacing(0.5),
    width: '30px',
    height: '30px',
  },
  dateStepsContainer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    padding: '1rem 0',
  },
});

export default styles;
