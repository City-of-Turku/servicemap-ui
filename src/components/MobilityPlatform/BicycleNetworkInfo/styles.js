const styles = {
  container: {
    margin: '0.5rem 1rem',
  },
  routeInfo: {
    border: '1px solid #000000',
    borderRadius: '5px',
    padding: '0.6rem',
    marginTop: '0.5rem',
  },
};

export default styles;
