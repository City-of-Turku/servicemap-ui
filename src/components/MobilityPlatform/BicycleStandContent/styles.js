const styles = {
  title: {
    marginBottom: '0.5rem',
    width: '85%',
    borderBottom: '1px solid #000000',
  },
};

export default styles;
