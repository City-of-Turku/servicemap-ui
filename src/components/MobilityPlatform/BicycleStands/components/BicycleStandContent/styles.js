export default theme => ({
  title: {
    marginBottom: theme.spacing(1),
    width: '85%',
    borderBottom: '1px solid #000000',
  },
  titleText: {
    fontSize: '0.9rem',
  },
  paragraph: {
    marginBottom: theme.spacing(0.5),
  },
  padding: {
    padding: theme.spacing(1),
  },
});
