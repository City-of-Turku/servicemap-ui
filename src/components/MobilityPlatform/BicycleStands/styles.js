const styles = {
  popupInner: {
    margin: '0.8rem',
  },
};

export default styles;
