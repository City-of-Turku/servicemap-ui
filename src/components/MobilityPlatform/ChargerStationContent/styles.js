const styles = {
  container: {
    margin: '0.8rem',
  },
  headerContainer: {
    width: '85%',
    borderBottom: '1px solid #000',
    paddingBottom: '0.5rem',
  },
  headerText: {
    marginBottom: '0.4rem',
  },
  textContainer: {
    marginTop: '0.7rem',
  },
};

export default styles;
