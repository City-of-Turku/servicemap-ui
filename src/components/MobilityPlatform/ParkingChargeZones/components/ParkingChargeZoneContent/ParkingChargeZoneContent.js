import { Typography } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';

const ParkingChargeZoneContent = ({ classes, intl, parkingChargeZone }) => {
  const renderText = (msgId, objValue, isTitle) => (
    <div className={isTitle ? classes.subtitle : classes.text}>
      <Typography variant={isTitle ? 'subtitle1' : 'body2'}>
        {intl.formatMessage({
          id: msgId,
        })}
        :
        {' '}
        {objValue}
      </Typography>
    </div>
  );

  return (
    <div className={classes.padding}>
      {renderText('mobilityPlatform.content.parkingChargeZones.zone', parkingChargeZone.extra.maksuvyohyke, true)}
      {renderText('mobilityPlatform.content.parkingChargeZones.price.weekDays', parkingChargeZone.extra.maksullisuus_arki, false)}
      {renderText('mobilityPlatform.content.parkingChargeZones.price.saturday', parkingChargeZone.extra.maksullisuus_lauantai, false)}
      {renderText('mobilityPlatform.content.parkingChargeZones.price.sunday', parkingChargeZone.extra.maksullisuus_sunnuntai, false)}
      {renderText('mobilityPlatform.content.parkingChargeZones.price', parkingChargeZone.extra.maksuvyohykehinta, false)}
    </div>
  );
};

ParkingChargeZoneContent.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  parkingChargeZone: PropTypes.objectOf(PropTypes.any),
};

ParkingChargeZoneContent.defaultProps = {
  parkingChargeZone: null,
};

export default ParkingChargeZoneContent;
