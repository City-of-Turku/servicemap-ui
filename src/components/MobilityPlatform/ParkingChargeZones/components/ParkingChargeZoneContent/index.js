import { withStyles } from '@mui/styles';
import { injectIntl } from 'react-intl';
import ParkingChargeZoneContent from './ParkingChargeZoneContent';
import styles from './styles';

export default withStyles(styles)(injectIntl(ParkingChargeZoneContent));
