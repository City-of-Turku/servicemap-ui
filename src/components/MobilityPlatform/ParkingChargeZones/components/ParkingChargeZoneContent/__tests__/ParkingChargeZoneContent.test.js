// Link.react.test.js
import React from 'react';
import { ThemeProvider } from '@mui/material/styles';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import themes from '../../../../../../themes';
import ParkingChargeZoneContent from '../index';
import finnishTranslations from '../../../../../../i18n/fi';

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnishTranslations,
};

const mockProps = {
  parkingChargeZone: {
    extra: {
      maksuvyohyke: '1',
      maksullisuus_arki: '9 - 18',
      maksullisuus_lauantai: '9 - 15',
      maksullisuus_sunnuntai: 'Maksuton',
      maksuvyohykehinta: '3 €/h',
    },
  },
};

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => (
  <IntlProvider {...intlMock}>
    <ThemeProvider theme={themes.SMTheme}>
      {children}
    </ThemeProvider>
  </IntlProvider>
);

const renderWithProviders = component => render(component, { wrapper: Providers });

describe('<ParkingChargeZoneContent />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<ParkingChargeZoneContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<ParkingChargeZoneContent {...mockProps} />);

    const p = container.querySelectorAll('p');
    const h6 = container.querySelector('h6');
    expect(h6.textContent).toEqual(`Vyöhyke: ${mockProps.parkingChargeZone.extra.maksuvyohyke}`);
    expect(p[0].textContent).toEqual(`Maksullisuus arkisin: ${mockProps.parkingChargeZone.extra.maksullisuus_arki}`);
    expect(p[1].textContent).toEqual(`Maksullisuus lauantaisin: ${mockProps.parkingChargeZone.extra.maksullisuus_lauantai}`);
    expect(p[2].textContent).toEqual(`Maksullisuus sunnuntaisin: ${mockProps.parkingChargeZone.extra.maksullisuus_sunnuntai}`);
    expect(p[3].textContent).toEqual(`Hinta: ${mockProps.parkingChargeZone.extra.maksuvyohykehinta}`);
  });
});
