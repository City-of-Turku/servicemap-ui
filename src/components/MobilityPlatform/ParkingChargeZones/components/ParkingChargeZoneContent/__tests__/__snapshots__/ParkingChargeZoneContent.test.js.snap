// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParkingChargeZoneContent /> should work 1`] = `
<div>
  <div
    class="injectIntl(ParkingChargeZoneContent)-padding-1"
  >
    <div
      class="injectIntl(ParkingChargeZoneContent)-subtitle-2"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
      >
        Vyöhyke
        :
         
        1
      </h6>
    </div>
    <div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Maksullisuus arkisin
        :
         
        9 - 18
      </p>
    </div>
    <div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Maksullisuus lauantaisin
        :
         
        9 - 15
      </p>
    </div>
    <div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Maksullisuus sunnuntaisin
        :
         
        Maksuton
      </p>
    </div>
    <div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Hinta
        :
         
        3 €/h
      </p>
    </div>
  </div>
</div>
`;
